include/diaspora.php

```diff
@@ -161,6 +162,56 @@ private static function verify_magic_envelope($envelope) {
 	}
 
 	/**
+	 * @brief encrypts data via AES
+	 *
+	 * @param string $key The AES key
+	 * @param string $iv The IV (is used for CBC encoding)
+	 * @param string $data The data that is to be encrypted
+	 *
+	 * @return string encrypted data
+	 */
```

Comment:
Have been using this with Diaspora for about the last 1.5 years without issue. 

You also *may* be able to replace the RINO compat function with openssl using 'aes-128-ecb' and 16-byte padding for both key and iv, but I couldn't say for sure as I haven't tested it. For the iv parameter, you'd probably need something like 

    openssl_random_pseudo_bytes(16,$placeholder);